beacon-chain/db/kv/wss.go

```diff
@@ -50,11 +39,24 @@ func (s *Store) SaveOrigin(ctx context.Context, serState, serBlock []byte) error
 	}
 	blk := wblk.Block()
 
-	// save block
 	blockRoot, err := blk.HashTreeRoot()
 	if err != nil {
 		return errors.Wrap(err, "could not compute HashTreeRoot of checkpoint block")
 	}
+
+	pr := blk.ParentRoot()
```

Comment:
if there is a node that is already checkpoint synced, is there a way they can trigger the backfill ? Or is this only possible for newly synced nodes running this PR

Reply:
`recoverLegacy` should handle nodes that were previously checkpoint synced. But now that you mention it, `recoverLegacy` has unit test coverage, but I have not manually tested checkpoint syncing a node on develop, and then restarting it from this branch to see what happens. I will try that out, and we should have nodes in our dev and prod clusters that were previously started on checkpoint sync where we can try it out.